sycl/test-e2e/AtomicRef/add.h

```diff
@@ -537,5 +537,14 @@ template <access::address_space space> void add_test_all() {
     add_test_orders_scopes<space, char *, ptrdiff_t>(q, N);
   }
 #endif
+
+#ifdef FULL_ATOMIC16_COVERAGE
```

Comment:
I haven't done `grep`, but a quick look suggests that we have no tests that define that macro, meaning that we never test this functionality. Perhaps it would be better to have `add-16.cpp` which would specifically test `half` and wouldn't be hidden under any `#ifdef`s